app/controllers/sentences_controller.php

```diff
@@ -773,6 +776,35 @@ public function search()
             $tags = implode(',', $tagsArray);
         }
 
+        // filter by list
+        $searchableLists = $this->SentencesList->getSearchableLists();
+        if (!empty($list)) {
+            $isSearchable = $this->SentencesList->isSearchableList($list);
+            if ($isSearchable) {
+                $sphinx['filter'][] = array('lists_id', $list);
+                $found = false;
+                foreach ($searchableLists as $rec) {
+                    if ($list == $rec['SentencesList']['id']) {
+                        $found = true;
+                        break;
+                    }
+                }
+                if (!$found) {
+                    $searchableLists[] = $isSearchable;
+                }
+            } else {
+                $ignored[] = format(
+                    /* @translators: This string will be preceded by
+                       “Warning: the following criteria have been
+                       ignored:” */
+                    __("“belongs to list number {listId}”, because this list does ".
```

Comment:
It is displayed that way because there may be more than one ignored criterion. See [this example](https://tatoeba.org/jpn/sentences/search?query=always&from=und&to=und&orphans=no&unapproved=no&native=yes&user=nonexsitant+user&tags=nonexistant+tag&has_audio=&trans_filter=limit&trans_to=und&trans_link=&trans_user=&trans_orphan=&trans_unapproved=&trans_has_audio=&sort=words).